pulsar-functions/worker/src/main/java/org/apache/pulsar/functions/worker/rest/api/ComponentImpl.java

```diff
@@ -1694,9 +1694,13 @@ public boolean isAuthorizedRole(String tenant, String namespace, String clientRo
             }
 
             if (clientRole != null) {
-                TenantInfo tenantInfo = worker().getBrokerAdmin().tenants().getTenantInfo(tenant);
-                if (tenantInfo.getAdminRoles() != null && tenantInfo.getAdminRoles().contains(clientRole)) {
-                    return true;
+                try {
+                    TenantInfo tenantInfo = worker().getBrokerAdmin().tenants().getTenantInfo(tenant);
+                    if (tenantInfo != null && tenantInfo.getAdminRoles() != null && tenantInfo.getAdminRoles().contains(clientRole)) {
+                        return true;
+                    }
+                } catch (PulsarAdminException.NotFoundException e) {
+
```

Comment:
dont really understand whats the intention here? if the tenant doesn't exist, won';t it fail in the next call for getting namespaces in that tenant?

Reply:
are you referring ```NamespaceName.get(tenant, namespace)```?  That just returns the correctly formatted name of the namespace

Reply:
Got it